openpype/modules/sync_server/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -92,86 +94,55 @@ def execute(self):
         anatomy = self.data.get("anatomy")
 
-        # Check it can proceed
-        if not project_doc and not asset_doc:
-            return
-
-        # Get subset id
-        subset_id = next(
-            (
-                subset["_id"]
-                for subset in get_subsets(
-                    project_name,
-                    asset_ids=[asset_doc["_id"]],
-                    fields=["_id", "data.family", "data.families"],
-                )
-                if subset["data"].get("family") == "workfile"
-                # Legacy compatibility
-                or "workfile" in subset["data"].get("families", {})
-            ),
-            None,
-        )
-        if not subset_id:
-            self.log.debug(
-                'No any workfile for asset "{}".'.format(asset_doc["name"])
-            )
-            return
-
-        # Get workfile representation
-        last_version_doc = get_last_version_by_subset_id(
-            project_name, subset_id, fields=["_id"]
-        )
-        if not last_version_doc:
-            self.log.debug("Subset does not have any versions")
-            return
+        context_filters = {
+            "asset": asset_name,
+            "family": "workfile",
+            "task": {"name": task_name, "type": task_type}
+        }
 
-        workfile_representation = next(
-            (
-                representation
-                for representation in get_representations(
-                    project_name, version_ids=[last_version_doc["_id"]]
-                )
-                if representation["context"]["task"]["name"] == task_name
-            ),
-            None,
-        )
+        workfile_representations = list(get_representations(
+            project_name,
+            context_filters=context_filters
+        ))
 
-        if not workfile_representation:
+        if not workfile_representations:
             self.log.debug(
                 'No published workfile for task "{}" and host "{}".'.format(
                     task_name, host_name
                 )
             )
             return
 
-        local_site_id = get_local_site_id()
-        sync_server.add_site(
-            project_name,
-            workfile_representation["_id"],
-            local_site_id,
-            force=True,
-            priority=99,
-            reset_timer=True,
-        )
+        sorted_workfile_representations = sorted(workfile_representations,
+                                                 key=lambda d: d["context"]
+                                                                ["version"])
 
-        while not sync_server.is_representation_on_site(
-            project_name, workfile_representation["_id"], local_site_id
-        ):
-            sleep(5)
-
-        # Get paths
-        published_workfile_path = get_representation_path(
-            workfile_representation, root=anatomy.roots
+        workfile_representation = sorted_workfile_representations[-1]
```

Comment:
This could also be `max(workfile_representations, key=lambda d: d["context"]["version"])`